src/us_congressional_districts/calibrate_with_microcalibrate.py

```diff
@@ -0,0 +1,147 @@
+from us_congressional_districts.calibrate import (
+    create_district_metric_matrix,
+    get_dataset,
+    create_target_matrix,
+    get_data_directory,
+)
+import pandas as pd
+import numpy as np
+import torch
+
+
+def create_households(
+    sample_per_district: int, data_by_household: pd.DataFrame
+):
+    synth_households = pd.DataFrame()
+    for district in age_data_by_district.index:
+        households_in_district = pd.DataFrame(
+            {
+                "household_id": data_by_household.sample(
+                    sample_per_district
+                ).index.values,
+            }
+        )
+        households_in_district["district"] = district
+        synth_households = pd.concat(
+            [synth_households, households_in_district]
+        )
+
+    return synth_households
+
+
+def abbrev_name(name):
+    # 'Congressional District 1 (118th Congress), Alabama' -> AL-01
+    district_number = name.split("District ")[1].split(" ")[0]
+    state = name.split(", ")[-1].strip()
+    return f"{state}{district_number.zfill(2)}".replace("-", "").replace(
+        "(at", "01"
+    )
+
+
+def create_target_names(
+    targets_by_district: pd.DataFrame, district_names: np.ndarray
+) -> list:
+    targets = []
+    for district in district_names:
+        for age_band in targets_by_district.columns:
+            targets.append(f"{district}/{age_band}")
+    return np.array(targets)
+
+
+age_data_by_district = pd.read_csv(
+    get_data_directory() / "input" / "demographics" / "age_district.csv"
+)
+
+target_state_names = age_data_by_district.NAME.copy().apply(abbrev_name)
+
+data_by_household = create_district_metric_matrix(
+    dataset=get_dataset("cps_2023", 2023),
+    ages=age_data_by_district,
+    time_period=2023,
+)
+
+targets_by_district = create_target_matrix(age_data_by_district)
+count_districts, count_targets = targets_by_district.shape
+target_names = create_target_names(targets_by_district, target_state_names)
+
+households = create_households(
+    sample_per_district=1_000, data_by_household=data_by_household
+)
```

Comment:
It's notable that all households do appear in the sample above. I'm not sure if that's guaranteed mathematically but it's good to see. It looks like households appear in 21 districts on average, with a spread of 5 to 40.

```
In [31]: data_by_household.shape
Out[31]: (20655, 18)

In [32]: len(set(households.household_id))
Out[32]: 20655
```

Reply:
Yes definitely not guaranteed- good to note I agree.